src/NimBLEAdvertising.cpp

```diff
@@ -333,48 +337,37 @@ void NimBLEAdvertising::start(uint32_t duration, void (*advCompleteCB)(NimBLEAdv
         }
 
         // check if there is room for the name, if not put it in scan data
-        if((payloadLen + m_advData.name_len) > 29) {
+        if((payloadLen + (2 + m_advData.name_len)) > BLE_HS_ADV_MAX_SZ) {
             if(m_scanResp){
                 m_scanData.name = m_advData.name;
                 m_scanData.name_len = m_advData.name_len;
+                if(m_scanData.name_len > BLE_HS_ADV_MAX_SZ - 2) {
+                    m_scanData.name_len = BLE_HS_ADV_MAX_SZ - 2;
```

Comment:
We need to add `m_scanData.name_is_complete = 0;` here. Since we are cutting off the name it should be marked incomplete.

Reply:
I'm sorry I'm incomplete.
I just fixed it

If you find another bug, you can fix it.